docs/architecture/swagger.v2.yaml

```
@@ -0,0 +1,636 @@
+                  links:
+                    type: object
+                    properties:
+                      api-docs:
+                        type: string
+                        format: uri
+                        example: 'https://opg-data-api.service.gov.uk/developer/'
+                  meta:
+                    type: object
+        '400':
+          description: Generic bad request (generally invalid syntax)
+          content:
+            application/vnd.opg-data.v1+json:
+              schema: 
+                $ref: '#/components/schemas/Error400'
+        '401':
+          description: Unauthorised (no current user and there should be)
+          content:
+            application/vnd.opg-data.v1+json:
+              schema: 
+                $ref: '#/components/schemas/Error401'
+        '403':
+          description: Forbidden - The current user is forbidden from accessing this data (in this way)
+          content:
+            application/vnd.opg-data.v1+json:
+              schema: 
+                $ref: '#/components/schemas/Error403'
+        '404':
+          description: That URL is not a valid route, or the item resource does not exist
+          content:
+            application/vnd.opg-data.v1+json:
+              schema: 
+                $ref: '#/components/schemas/Error404'
+        '413':
+          description: Payload too large
+          content:
+            application/vnd.opg-data.v1+json:
+              schema: 
+                $ref: '#/components/schemas/Error413'
+        '415':
+          description: Unsupported media type
+          content:
+            application/vnd.opg-data.v1+json:
+              schema: 
+                $ref: '#/components/schemas/Error415'
+        '500':
+          description: Something unexpected happened and it is the API's fault
+          content:
+            application/vnd.opg-data.v1+json:
+              schema: 
+                $ref: '#/components/schemas/Error500'
+        '503':
+          description: Service Unavailable - please try again later
+          content:
+            application/vnd.opg-data.v1+json:
+              schema: 
+                $ref: '#/components/schemas/Error503'
+  '/reports/{id}/supportingdocuments':
+    description: Supporting documents associated with client reports
+    post:
+      summary: Add a document to Supervision which supports a report
+      description: Add a document to Supervision which supports a report
+      operationId: addReportSupportingDocument
+      tags:
+        - 'deputy-reporting'
+      parameters:
+        - in: path
+          name: id
+          required: true
+          description: Report UUID
+          schema:
+            $ref: '#/components/schemas/ReportId'
+      requestBody:
+        required: true
+        description: The supporting document
+        content:
+          image/*:
+            schema:
+              type: string
+              format: binary
+          application/*:
```

Should this just be PDF? I'm not aware of other application subtypes we'd accept
```suggestion
          application/pdf:
```

---

Hey - this is for the supportingdocs endpoint, which I believe can be any sort of file that the user uploads?

---

Our restrictions are that it must be one of:
- application/pdf
- application/x-pdf
- image/png
- image/jpeg

I've no issue with the API being more liberal, but thought I should check.

---

That's good that it's so limited - I believe we'll be sending financial csvs through this endpoint as well now though?

---

Great point, that's `text/csv` also then